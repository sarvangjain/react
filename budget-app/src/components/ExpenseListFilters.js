import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';


export class ExpenseListFilters extends React.Component {
    
    state = {
        calendarFocused: null
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}));
    }

    inputOnChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onChange = (e) => {
        e.target.value === 'date' ? this.props.sortByDate() : this.props.sortByAmount();
    }

    render() {
        return (    
            <div>
                <input 
                    type='text' 
                    value={this.props.filters.text} 
                    onChange={this.inputOnChange}/>
                <select 
                    value={this.props.filters.sortBy} 
                    onChange={this.onChange}>
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
}

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (data) => dispatch(setTextFilter(data)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())
})
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);