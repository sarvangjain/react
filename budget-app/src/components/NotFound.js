import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <div align='center'>
        <p>Error 404!</p>
        <p>Page Not Found</p>
        <Link to='/'>Go to Home Page</Link>
    </div>

);

export default NotFoundPage;