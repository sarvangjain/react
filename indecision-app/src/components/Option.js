import React from 'react'

const Option = (props) => (
	<div className="option">
		<p className="option__text">{props.count}. {props.opt}</p>
		<button
			className="button button--link" 
			onClick={(e) => {
				props.handleDeleteOption(props.opt);
			}}
		>
			Remove
		</button>
	</div>
);
    
export default Option;