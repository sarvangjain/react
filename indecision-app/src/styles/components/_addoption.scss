.add-option-error {
    color: $off-white;
    font-style: italic;
    padding: 0 $m-size;
    margin: $m-size 0 0 0;
    text-align: center;
}

.add-option {
    display: flex;
    padding: $m-size;
}

.add-option__input{
    background: $dark-blue;
    border: none;
    border-bottom: 0.3rem solid darken($dark-blue, 10%);
    margin-right: $s-size;
    flex-grow: 1;
    color: $off-white;
    padding: $s-size;
}