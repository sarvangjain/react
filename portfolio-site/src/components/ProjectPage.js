import React from 'react';

const ProjectPage = (props) => (
    <div>
        <h1>Thing I have done:</h1>
        <p>This page is for the Project {props.match.params.id}</p>
    </div>
);

export default ProjectPage;