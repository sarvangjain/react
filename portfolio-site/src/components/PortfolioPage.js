import React from 'react';
import { NavLink } from 'react-router-dom';
import ProjectPage from './ProjectPage';

const PortfolioPage = () => (
    <div>
        <h1>My Projects</h1>
        <p>Check out the following project:</p>
        <div>
            <NavLink to="/portfolio/1" activeClassName="is-active">Project 1 </NavLink>
            <NavLink to="/portfolio/2" activeClassName="is-active">Project 2 </NavLink>
        </div>
    </div>
);

export default PortfolioPage;