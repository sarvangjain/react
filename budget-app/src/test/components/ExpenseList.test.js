import { ExpenseList } from '../../components/ExpenseList';
import ExpenseListItem from '../../components/ExpenseListItem';
import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import toJSON, { toJson } from 'enzyme-to-json';

test('should render ExpenseList with expense', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should render ExpenseList without expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});
