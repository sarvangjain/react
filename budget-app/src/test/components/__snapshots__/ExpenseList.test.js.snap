// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expense 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={232400}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={15000}
    createdAt={345600000}
    description="Water Bill"
    id="3"
    key="3"
    note=""
  />
  <ExpenseListItem
    amount={20000}
    createdAt={172800000}
    description="Gas Bill"
    id="4"
    key="4"
    note=""
  />
</div>
`;

exports[`should render ExpenseList without expenses 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
